textfsm.py

```diff
@@ -893,6 +893,35 @@ def ParseText(self, text, eof=True):
 
     return self._result
 
+  def ParseTextToDicts(self, *args, **kwargs):
+    """Calls ParseText and turns the result into list of dicts.
+
+    List items are dicts of rows, dict key is column header and value is column
+    value.
+
+    Args:
+      text: (str), Text to parse with embedded newlines.
+      eof: (boolean), Set to False if we are parsing only part of the file.
+            Suppresses triggering EOF state.
+
+    Raises:
+      TextFSMError: An error occurred within the FSM.
+
+    Returns:
+      List of dicts.
+    """
+
+    result_lists = self.ParseText(*args, **kwargs)
+    result_dicts = []
+
+    for row in result_lists:
+      result_dict = {}
+      for i in range(len(row)):
+        result_dict[self.header[i]] = row[i]
```

Comment:
You should use `zip`.
```py
for column_header, column_value in zip(self.header, row):
    result_dict[column_header] = column_value
```
or even more Pythonic:
```py
result_dict = dict(zip(self.header, row))
```
For instance:
```py
>>> header = ["name", "age", "city"]
>>> row = ["George", 76, "New York"]
>>> dict(zip(header,row))
{'name': 'George', 'age': 76, 'city': 'New York'}
```

Reply:
You could go even further (but you shouldn't, unless you think it's readable, then OK):
```py
>>> header = ["name", "age", "city"]
>>> rows = [["George", 76, "New York"], ["Judy", 49, "Munich"]]
>>> [dict(zip(row, header)) for row in rows]
[{'George': 'name', 76: 'age', 'New York': 'city'}, {'Judy': 'name', 49: 'age', 'Munich': 'city'}]
```